core/src/main/java/com/netease/arctic/catalog/CatalogLoader.java

```diff
@@ -114,22 +104,23 @@ public static ArcticCatalog load(AmsClient client, String catalogName, Map<Strin
       switch (type) {
         case CATALOG_TYPE_HADOOP:
           Preconditions.checkArgument(
-              tableFormat.equals(TableFormat.ICEBERG),
-              "Hadoop catalog support iceberg table only.");
-          if (catalogMeta.getCatalogProperties().containsKey(CatalogMetaProperties.TABLE_FORMATS)) {
+              TableFormat.ICEBERG == tableFormat || TableFormat.MIXED_ICEBERG == tableFormat,
+              "Hadoop catalog support iceberg/mixed-iceberg table only.");
+          if (TableFormat.ICEBERG == tableFormat) {
             catalogImpl = ICEBERG_CATALOG_IMPL;
           } else {
-            // Compatibility with older versions
-            catalogImpl = AMS_CATALOG_IMPL;
+            catalogImpl = MIXED_ICEBERG_CATALOG_IMP;
           }
           break;
         case CATALOG_TYPE_HIVE:
           if (tableFormat.equals(TableFormat.ICEBERG)) {
             catalogImpl = ICEBERG_CATALOG_IMPL;
           } else if (tableFormat.equals(TableFormat.MIXED_HIVE)) {
             catalogImpl = HIVE_CATALOG_IMPL;
+          } else if (TableFormat.MIXED_ICEBERG == tableFormat) {
```

Comment:
Although using both `==` and `equals` would work here, I prefer to keep the approach consistent.